ibm/service/backuprecovery/resource_ibm_backup_recovery_connector_access_token.go

```diff
@@ -0,0 +1,162 @@
+		}
+	}
+	return nil
+}
+
+func resourceIbmBackupRecoveryConnectorAccessTokenCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	backupRecoveryConnectorClient, err := meta.(conns.ClientSession).BackupRecoveryV1Connector()
+	if err != nil {
+		tfErr := flex.DiscriminatedTerraformErrorf(err, err.Error(), "ibm_backup_recovery_connector_access_token", "create", "initialize-client")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
+	}
+
+	if backupRecoveryConnectorClient.GetConnectorURL() == "" {
+		tfErr := flex.DiscriminatedTerraformErrorf(nil, "No connector URL specified. Please set the `IBMCLOUD_BACKUP_RECOVERY_CONNECTOR_ENDPOINT` environment variable or specify the endpoint in endpoints.json file.", "ibm_backup_recovery_connector_access_token", "create", "initialize-client")
+		return tfErr.GetDiag()
+	}
+
+	createAccessTokenOptions := &backuprecoveryv1.CreateAccessTokenOptions{}
+
+	if _, ok := d.GetOk("username"); ok {
+		createAccessTokenOptions.SetUsername(d.Get("username").(string))
+	}
+	if _, ok := d.GetOk("password"); ok {
+		createAccessTokenOptions.SetPassword(d.Get("password").(string))
+	}
+	if _, ok := d.GetOk("domain"); ok {
+		createAccessTokenOptions.SetDomain(d.Get("domain").(string))
+	}
+
+	result, _, err := backupRecoveryConnectorClient.CreateAccessTokenWithContext(context, createAccessTokenOptions)
+	if err != nil {
+		tfErr := flex.TerraformErrorf(err, fmt.Sprintf("CreateAccessTokenWithContext failed: %s", err.Error()), "ibm_backup_recovery_connector_access_token", "read")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
+	}
+
+	d.SetId(resourceIbmBackupRecoveryConnectionAccessTokenID(d))
+
+	if !core.IsNil(result.AccessToken) {
+		if err = d.Set("access_token", result.AccessToken); err != nil {
+			err = fmt.Errorf("Error setting access_token: %s", err)
+			return flex.DiscriminatedTerraformErrorf(err, err.Error(), "ibm_backup_recovery_connector_access_token", "read", "set-access_token").GetDiag()
+		}
+	}
+	if !core.IsNil(result.Privileges) {
+		if err = d.Set("privileges", result.Privileges); err != nil {
+			err = fmt.Errorf("Error setting privileges: %s", err)
+			return flex.DiscriminatedTerraformErrorf(err, err.Error(), "ibm_backup_recovery_connector_access_token", "read", "set-privileges").GetDiag()
+		}
+	}
+	if !core.IsNil(result.TokenType) {
+		if err = d.Set("token_type", result.TokenType); err != nil {
+			err = fmt.Errorf("Error setting token_type: %s", err)
+			return flex.DiscriminatedTerraformErrorf(err, err.Error(), "ibm_backup_recovery_connector_access_token", "read", "set-token_type").GetDiag()
+		}
+	}
+
+	return resourceIbmBackupRecoveryConnectorAccessTokenRead(context, d, meta)
+}
+
+func resourceIbmBackupRecoveryConnectionAccessTokenID(d *schema.ResourceData) string {
+	return time.Now().UTC().String()
```

Comment:
Setting ID as time is good for datasources n't for resources
Can we set with any argument which is required?